### PR TITLE
Add some utils functions  and lua support

### PR DESCRIPTION
**Pull request by bintester**
Which branch your pull-request should merge into?

- `2.1`: BugFixes and Improvements for Current LTS branch
- `dev`: New features or remove deprecated features

## Describe your changes
UIButton bugfix
add some utils functions  and lua support

### For each PR
- [ ] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [x] I have performed a self-review of my code.
       
   Optional:
   - [x] I have checked readme and add important infos to this PR.
   - [x] I have added/adapted some tests too.
          
### For core/new feature PR
- [ ] I have checked readme and add important infos to this PR.
- [ ] I have added thorough tests.

### CODE REVIEW

File: core/base/Utils.cpp [hunk 1]
```diff
@@ -414,6 +414,14 @@ std::string getDataMD5Hash(const Data& data)
     return computeDigest(std::string_view{(const char*)data.getBytes(), (size_t)data.getSize()}, "md5"sv);
 }
 
+std::string getStringMD5Hash(std::string_view str)
+{
+    if (str.empty())
```
Comment (halx99) [inline core/base/Utils.cpp]: empty message also can return a valid digest

Comment (bintester) [reply core/base/Utils.cpp]: I modified some code and tested "MD5", "SHA1", "SHA256", "SHA512",please review

**Review by halx99 — COMMENTED** [review]
https://github.com/axmolengine/axmol/pull/1629/files#diff-6838aab08f6d69433e4877011bca8ef7fbbd993a13532b7f2592863f5dc7c7a2R439 please help check at here:
```cpp
if (!data.empty()) {
     EVP_DigestUpdate(mdctx, data.data(), data.size());
}
```

File: core/base/Utils.cpp (hunk 1 above)
Comment (halx99) [reply core/base/Utils.cpp]: nice

**Pull request merged**